### PR TITLE
[move-2024] Update unit testing doc

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Updates docs for unit testing in Move for Move 2024.

### CODE REVIEW

File: external-crates/move/documentation/book/src/unit-testing.md [hunk 1]
```diff
@@ -12,29 +12,30 @@ annotations can be placed on a function with any visibility. Whenever a module o
...
+fun this_is_not_correct(arg: u64) { ... }
 ```
 
-A test can also be annotated as an `#[expected_failure]`. This annotation marks that the test is
-expected to raise an error. You can ensure that a test is aborting with a specific abort code by
-annotating it with `#[expected_failure(abort_code = <code>)]`, if it then fails with a different
-abort code or with a non-abort error the test will fail. Only functions that have the `#[test]`
-annotation can also be annotated as an #`[expected_failure]`.
+A test can also be annotated as an `#[expected_failure]`. This annotation marks
+that the test is expected to raise an error. There are a number of
```
Comment (tnowacki) [inline external-crates/move/documentation/book/src/unit-testing.md:29]: The wrapping here seems a bit strange, do you have prettier installed?

File: external-crates/move/documentation/book/src/unit-testing.md [hunk 2]
```diff
@@ -43,96 +44,235 @@ public fun this_test_will_abort_and_pass() { abort 1 }
...
-#[test(arg = @0xC0FFEE)] // OK
-fun this_is_correct_now(arg: signer) { ... }
 
-#[test(wrong_arg_name = @0xC0FFEE)] // Not correct: arg name doesn't match
-fun this_is_incorrect(arg: signer) { ... }
+### 2. `#[expected_failure(abort_code = <constant>)]`
+
+This will pass if the test
+aborts with the specified constant value in the module that defines the
+constant, and fail otherwise.
```
Comment (tnowacki) [inline external-crates/move/documentation/book/src/unit-testing.md:82]: We should highlight immediately that this is the recommended way for most tests. And maybe even reorder these items to stress that?

**Review by tnowacki — APPROVED** [review]

File: external-crates/move/documentation/book/src/unit-testing.md (hunk 1 above)
Comment (tzakian) [reply external-crates/move/documentation/book/src/unit-testing.md:29]: What is prettier? I just use `vip gq` in Vim... :p

Comment (tzakian) [reply external-crates/move/documentation/book/src/unit-testing.md:29]: Looks like the formatting got messed up during a rebase :'(

**Pull request merged**